src/custom-element.js

```diff
@@ -1429,9 +1440,44 @@ function createBaseAmpElementProto(win) {
 export function registerElement(win, name, implementationClass) {
   knownElements[name] = implementationClass;
 
-  win.document.registerElement(name, {
-    prototype: createAmpElementProto(win, name),
-  });
+  const supportsCustomElementsV1 = 'customElements' in win;
+  if (supportsCustomElementsV1) {
+    // `customElements.define` requires a function or ES2015 class to be passed
```

Comment:
Would it make sense to instead make the inheritance from HTMLElement conditional upon whether we are going for V1 or V0?